### PR TITLE
support for multiple video streams

### PR DESCRIPTION
**Pull request by mcosmin222**
This PR is to add support for multiple video streams.

I did not have any file to check if this actually works, so it might need some more work.
It also broke the frame grabber, we need to think of how it will support multiple video streams. 

One approach is to pass the stream from which it is going to grab the frame as parameter to the method. Another approach would be to give off a blind index for the stream.

### CODE REVIEW

**Comment by lukasf** [comment]
Wow that was fast! One general remark: I think we should not introduce breaking changes in our main class. So I'd propose to leave the old "VideoStream" property as "Deprecated" and have it return the first video stream. 

Then we also do not need to change the FrameGrabber until we have a good idea how to select video streams there.

**Comment by mcosmin222** [comment]
I thought removing the deprecated metjods was the right thing to do at this time. I can surely bring it back.

Another idea is to allow creation of frame grabbers from a video stream info. We can clone the input stream and create a frame grabber from there.

**Comment by mcosmin222** [comment]
I will handle  this in the coming days.

**Comment by mcosmin222** [comment]
Actually, the fix was far easier. 

Now the only way in which the frame grabber is broken is because we can't select the stream. It will automatically select the first or the default stream.

**Comment by lukasf** [comment]
The VideoStream property was not deprecated, so we should not remove it just like that. Just have it return the first video stream (or null, if none was found).

Right now we can only create the FrameGrabber from IRandomAccessStream. So it does not support DASH, and so I don't believe we will need video stream selection there. We can add this if someone really wants it, but right now I would not bother with that.

**Comment by mcosmin222** [comment]
Hmm yes you are right. It wasn't deprecated. Though i distinctly remember removing something deprecated. Sorry.

I will put it back.

**Comment by lukasf** [comment]
Oh, but maybe we should first test it? I will try to get this running with DASH later today or tomorrow latest.

File: FFmpegInterop/FFmpegInteropMSS.cpp [hunk 1]
```diff
@@ -816,18 +819,33 @@ HRESULT FFmpegInteropMSS::InitFFmpegContext()
...
+			if (stream)
 			{
-				auto pixelAspect = (double)VideoDescriptor->EncodingProperties->PixelAspectRatio->Numerator / VideoDescriptor->EncodingProperties->PixelAspectRatio->Denominator;
-				auto videoAspect = ((double)videoStream->m_pAvCodecCtx->width / videoStream->m_pAvCodecCtx->height) / pixelAspect;
+				auto streamDescriptor = dynamic_cast<VideoStreamDescriptor^>(stream->StreamDescriptor);
+				auto pixelAspect = (double)streamDescriptor->EncodingProperties->PixelAspectRatio->Numerator / streamDescriptor->EncodingProperties->PixelAspectRatio->Denominator;
+				auto videoAspect = ((double)stream->m_pAvCodecCtx->width / stream->m_pAvCodecCtx->height) / pixelAspect;
 
-				videoStreamInfo = ref new VideoStreamInfo(stream->Name, stream->Language, stream->CodecName, avStream->codecpar->bit_rate, true,
+				auto info = ref new VideoStreamInfo(stream->Name, stream->Language, stream->CodecName, avStream->codecpar->bit_rate, true,
```
Comment (lukasf) [inline FFmpegInterop/FFmpegInteropMSS.cpp]: Pass "isDefault" here instead of hard coded "true"

File: FFmpegInterop/FFmpegInteropMSS.cpp [hunk 2]
```diff
@@ -1525,6 +1557,12 @@ void FFmpegInteropMSS::OnSwitchStreamsRequested(MediaStreamSource ^ sender, Medi
 		currentAudioStream->DisableStream();
 		currentAudioStream = nullptr;
 	}
+	if (currentVideoStream && args->Request->OldStreamDescriptor == currentVideoStream->StreamDescriptor)
+	{
+		currentAudioStream->DisableStream();
```
Comment (lukasf) [inline FFmpegInterop/FFmpegInteropMSS.cpp]: Must be currentVideoStream

**Comment by mcosmin222** [comment]
That's an unfortunate turn of events. I distinctly remember seeing a track selection in one of the samples last year but maybe it was custom transport controls. We could do the same here.

**Pull request merged**